app/src/main/kotlin/com/alexetnico/rxvscoroutines/ui/MainActivity.kt

```diff
@@ -17,27 +18,30 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        observeViewModel()
+        random_button.setOnClickListener { onClickRandomBeer() }
     }
 
-    private fun observeViewModel() {
+    private fun onClickRandomBeer() {
 
-        viewModel.beersCo.observe(this, Observer {
+        viewModel.beerCo.observe(this, Observer {
             it?.let {
-                Log.d("BEER", "$it")
-            }
-        })
-
-        viewModel.beersRx.observe(this, Observer {
-            it?.let {
-                toast(it.first().name)
+                random_beer_co.setup(it)
             }
         })
 
         viewModel.beerRx.observe(this, Observer {
             it?.let {
-                toast(it.name)
+                random_beer_rx.setup(it)
             }
         })
     }
+
+    private fun BeerCard.setup(beer: Beer) = this.setupView(
```

Comment:
mdr tu t'es cru chaud en faisant une extension avoue ???
Sauf que la ca sert à R non ? Tu fais direct `random_beer_rx.setupView(it)`narvalos ;)